app/controllers/main/volunteerRegisterEvents.py

```diff
@@ -0,0 +1,67 @@
+from app.models.event import Event
+from app.models.user import User
+from app.models.eventParticipant import EventParticipant
+from app.controllers.admin import admin_bp
+from app.controllers.events import events_bp
+from app.logic.userRsvpForEvent import userRsvpForEvent, unattendedRequiredEvents
+from flask import flash, request, redirect, url_for, g
+
+
+
+@admin_bp.route('/rsvpForEvent', methods=['POST'])
+def volunteerRegister():
+    """
+    This function selects the user ID and event ID and registers the user
+    for the event they have clicked register for.
+    """
+    eventData = request.form
+    userId = User.get(User.username == g.current_user)
+    eventId = eventData['eventId']
+    program = eventData['programId']
+    isEligible = userRsvpForEvent(userId, eventId)
+    listOfRequirements = unattendedRequiredEvents(program, userId)
+
+
+    if not isEligible: # if they are banned
+        flash(f"Cannot RSVP: {userId.firstName} is banned")
+
+    elif len(listOfRequirements) > 0:

```

Comment:
We should talk about this kind of warning. Because the trainings aren't required anymore.